meshtastic/tests/test_mesh_interface.py

```
@@ -3557,6 +3573,53 @@ def pop(  # type: ignore[override]
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """A packet should be requeued when the send path raises before successful handoff."""
+    with MeshInterface(noProto=True) as iface:
+        iface.noProto = False
+        packet = mesh_pb2.ToRadio()
+        packet.packet.id = 123
+        incoming = mesh_pb2.ToRadio()
+        incoming.packet.id = 999
+
+        def _failing_send(_msg: mesh_pb2.ToRadio) -> None:
+            raise RuntimeError("send failed")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)